sklearn/datasets/samples_generator.py

```diff
@@ -28,8 +27,7 @@ def _generate_hypercube(samples, dimensions, rng):
     if dimensions > 30:
         return np.hstack([_generate_hypercube(samples, dimensions - 30, rng),
                           _generate_hypercube(samples, 30, rng)])
-    out = astype(sample_without_replacement(2 ** dimensions, samples,
-                                            random_state=rng),
+    out = astype(random_state.permutation(2 ** dimensions)[:samples],
```

Comment:
This test: https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/decomposition/tests/test_pca.py#L194

Which I found the connection to this method: https://github.com/scikit-learn/scikit-learn/pull/7703#issuecomment-255006786